src/components/CameraControls/index.tsx

```diff
@@ -1,26 +1,152 @@
-import * as React from "react";
+import React, { useEffect, useState, useRef } from "react";
 import { Button, Tooltip } from "antd";
+import classNames from "classnames";
+import { useHotkeys } from "react-hotkeys-hook";
 
 import { TOOLTIP_COLOR } from "../../constants/index";
 import Icons from "../Icons";
 
 const styles = require("./style.css");
 
+const PAN = "pan";
+const ROTATE = "rotate";
+const CAMERA_MODE_MODIFIER_KEYS = ["Meta", "Shift"];
+const ZOOM_IN_HK = "ArrowUp";
+const ZOOM_OUT_HK = "ArrowDown";
+const RESET_HK = "r";
+const HOT_KEYS = [ZOOM_IN_HK, ZOOM_OUT_HK, RESET_HK];
+
 interface CameraControlsProps {
     resetCamera: () => void;
     zoomIn: () => void;
     zoomOut: () => void;
+    setPanningMode: (value: boolean) => void;
 }
 
 const CameraControls = ({
     resetCamera,
     zoomIn,
     zoomOut,
+    setPanningMode,
 }: CameraControlsProps) => {
+    const [mode, setMode] = useState(ROTATE);
+    const [keyPressed, setKeyPressed] = useState("");
+    const lastKeyPressed = useRef("");
+    const isModifierKey = (key: string) =>
+        CAMERA_MODE_MODIFIER_KEYS.includes(key);
+    useHotkeys(
```

Comment:
[nit] A blank line between lines 36 and 37 would be good; I got confused for a bit thinking `useHotkeys` was a part of `isModifierKey` and that we were missing brackets and indentation 😅